### PR TITLE
Fix AccummulatorCompiler to work with type-specific stack representation

### PR DESCRIPTION
**Pull request by losipiuk**
Fix aggregation compiler to work fine if input function is defined for
Type which uses non standard stack representation. E.g. TIMESTAMP(p>6)
uses LongTimestamp on stack.

### CODE REVIEW

**Review by findepi — APPROVED** [review]
LGTM but please add a test, or just change existing place(s) to leverage that.

**Review by dain — APPROVED** [review]

**Comment by losipiuk** [comment]
rebase

**Comment by losipiuk** [comment]
@findepi  test added

File: presto-main/src/test/java/io/prestosql/operator/aggregation/TestAccumulatorCompiler.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        // test if we can compile aggregation
+        assertThat(AccumulatorCompiler.generateAccumulatorFactoryBinder(metadata, classLoader)).isNotNull();
+
+        // TODO test if aggregation actually works...
+    }
+
+    public interface TestAggregationState
+            extends AccumulatorState
+    {
+    }
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/operator/aggregation/TestAccumulatorCompiler.java:78]:
```suggestion
    public interface TestAggregationState
            extends AccumulatorState {}
```

File: presto-main/src/test/java/io/prestosql/operator/aggregation/TestAccumulatorCompiler.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    @ReportUnannotatedMethods.Suppress
+    public static void longTimestampAggregationInput(TestAggregationState state, LongTimestamp value) {}
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/operator/aggregation/TestAccumulatorCompiler.java:81]: Any existing aggs that can benefit from your change?

File: presto-main/src/test/java/io/prestosql/operator/aggregation/TestAccumulatorCompiler.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+        // TODO test if aggregation actually works...
+    }
+
+    public interface TestAggregationState
+            extends AccumulatorState
+    {
+    }
+
+    @ReportUnannotatedMethods.Suppress
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/operator/aggregation/TestAccumulatorCompiler.java:80]: Just move those static methods into inner class (just as if you were declaring aggregation)

**Review by findepi — APPROVED** [review]

File: presto-main/src/test/java/io/prestosql/operator/aggregation/TestAccumulatorCompiler.java (hunk 2 above)
Comment (losipiuk) [reply presto-main/src/test/java/io/prestosql/operator/aggregation/TestAccumulatorCompiler.java:81]: nope :)

**Pull request merged**